src/main/scala/com/rbmhtechnology/calliope/serializer/SequencedEventSerializer.scala

```
@@ -61,10 +64,14 @@ abstract class SequencedEventSerializer(system: ExtendedActorSystem, payloadSeri
       .setCreationTimestamp(sequencedEvent.creationTimestamp.toEpochMilli)
 
   def sequencedEvent(format: SequencedEventFormat): SequencedEvent[_] =
-    SequencedEvent(
-      payloadSerializer.payload(format.getPayload),
-      format.getSourceId,
-      format.getSequenceNr,
-      Instant.ofEpochMilli(format.getCreationTimestamp)
-    )
+    try {
+      SequencedEvent(
+        payloadSerializer.payload(format.getPayload),
+        format.getSourceId,
+        format.getSequenceNr,
+        Instant.ofEpochMilli(format.getCreationTimestamp)
+      )
+    } catch {
+      case NonFatal(err) => throw new PayloadNotSerializableException("Unable to deserialize payload", err, format.getPayload)
```

Why to try/catch/throw here again? `PayloadSerializer.payload` already does that.

---

+1

---

Because the payload a different.

Given you have an Envelope with `PayloadFormat` wrapping a `SequencedEventFormat` wrapping a `AppPayload`. If you catch the exception only in `PayloadSerializer.payload` then the payload information in the Exception is about the `SequencedEventFormat` payload, catching it in the `SequencedEventSerializer` will actually result in information about the `AppPayload`.

---

I think we can (and should) safely assume backwards compatibility of our internal wrappers (i.e. protocol headers). We should not be able ignore errors in these cases.

---

And even if we encounter one, it's fine to report the first encountered one.

---

I think this is what this implementation does. It reports the first encountered error correctly and that is the `PayloadFormat` deserialization fails inside the `SequencedEvent`. 

I deemed it would be important to include the correct serializeID/manifest in the `PayloadNotSerializableException` so it is actually meaningful. Otherwise there will only be the id/manifest of the `SequencedEventSerializer` in the exception no matter what actual payload was in the `SequencedEvent`.

---

Keep it as simple as possible at the moment. We will later add generic support for protocol stacks where we need to rethink error handling and reporting. But if you're more comfortable with your current proposal/solution let's keep it as a temporary solution.

---

+1

Remove with latest commit.